lnwallet/chancloser/rbf_coop_transitions.go

```diff
@@ -0,0 +1,1102 @@
+		// process.
+		//
+		// TODO(roasbeef): need to be able to omit an output when
+		// signing based on the above, as closing opt
+		rawSig, _, _, err := env.CloseSigner.CreateCloseProposal(
+			msg.SigMsg.FeeSatoshis, l.LocalDeliveryScript,
+			l.RemoteDeliveryScript, chanOpts...,
+		)
+		if err != nil {
+			return nil, err
+		}
+		wireSig, err := lnwire.NewSigFromSignature(rawSig)
+		if err != nil {
+			return nil, err
+		}
+
+		localSig, err := wireSig.ToSignature()
+		if err != nil {
+			return nil, err
+		}
+
+		// With our signature created, we'll now attempt to finalize
+		// the close process.
+		//
+		closeTx, _, err := env.CloseSigner.CompleteCooperativeClose(
+			localSig, remoteSig, l.LocalDeliveryScript,
+			l.RemoteDeliveryScript, msg.SigMsg.FeeSatoshis,
+			chanOpts...,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		chancloserLog.Infof("ChannelPoint(%v): received sig (fee=%v "+
+			"sats) from remote party, signing new tx=%v",
+			env.ChanPoint, msg.SigMsg.FeeSatoshis,
+			lnutils.SpewLogClosure(closeTx),
+		)
+
+		var closingSigs lnwire.ClosingSigs
+		if noClosee {
+			closingSigs.CloserNoClosee = newSigTlv[tlv.TlvType1](wireSig)
+		} else {
+			closingSigs.CloserAndClosee = newSigTlv[tlv.TlvType3](wireSig)
+		}
+
+		// As we're about to broadcast a new version of the co-op close
+		// transaction, we'll mark again as broadcast, but with this
+		// variant of the co-op close tx.
+		//
+		// TODO(roasbeef): db will only store one instance, store both?
+		err = env.ChanObserver.MarkCoopBroadcasted(closeTx, false)
+		if err != nil {
+			return nil, err
+		}
+
+		// As we transition, we'll omit two events: one to broadcast
+		// the transaction, and the other to send our ClosingSig
+		// message to the remote party.
+		sendEvent := &protofsm.SendMsgEvent[ProtocolEvent]{
+			TargetPeer: env.ChanPeer,
+			Msgs: []lnwire.Message{&lnwire.ClosingSig{
+				ChannelID:   env.ChanID,
+				ClosingSigs: closingSigs,
+			}},
+		}
+		broadcastEvent := &protofsm.BroadcastTxn{
+			Tx: closeTx,
+			Label: labels.MakeLabel(
+				labels.LabelTypeChannelClose, &env.Scid,
+			),
+		}
+		daemonEvents := protofsm.DaemonEventSet{sendEvent, broadcastEvent}
+
+		// Now that we've extracted the signature, we'll transition to
+		// the next state where we'll sign+broadcast the sig.
+		return &CloseStateTransition{
+			NextState: &ClosePending{
+				CloseTx: closeTx,
+			},
+			NewEvents: fn.Some(protofsm.EmittedEvent[ProtocolEvent]{
+				ExternalEvents: fn.Some(daemonEvents),
+			}),
+		}, nil
+	}
+
+	return nil, fmt.Errorf("%w: received %T while in RemoteCloseStart",
+		ErrInvalidStateTransition, event)
+}
+
+// ProcessEvent is a semi-terminal state in the rbf-coop close state machine.
+// In this state, we're waiting for either a confirmation, or for either side
+// to attempt to create a new RBF'd co-op close transaction.
+func (c *ClosePending) ProcessEvent(event ProtocolEvent, env *Environment,
```

Comment:
ProcessEvent isn't a state at all. o.O

Reply:
I say `ClosePending` is semi-terminal here as it might just lead to `CloseFin` upon a confirm event, or one side sends a shutdown, then we go back to the top.